jax/numpy/__init__.py

```diff
@@ -459,6 +460,11 @@
         "jax.numpy.PZERO is deprecated. Use 0.0 instead.",
         0.0,
     ),
+    # Added Aug 17, 2023:
+    "issubsctype": (
+        "jax.numpy.issubsctype is deprecated. Use numpy.core.numerictypes.issubsctype instead.",
```

Comment:
Is this the right recommendation? Presumably you should want a public API, and doesn't this do what `issubdtype` does, roughly?

Reply:
We're removing this because `numpy.issubsctype` is being removed. I believe the path mentioned here is the new public path for that function.

Reply:
The decision about sctype related functions was that [All *sctype functions can be deprecated and removed](https://numpy.org/neps/nep-0052-python-api-cleanup.html#cleaning-up-the-main-namespace) so in numpy 2.0 they won't be available in the public API. Sorry for not stating it clearly: The path from the internal module would allow to provide it with `jnp.issubsctype`anyway. In case you decided to also drop it, then recommendation for end-user would be to not use it at all.

Reply:
Indeed, that was my read also: the user simply shouldn't be calling any variant of this function.

Reply:
@mtsokol let me repeat something I mentioned previously, though: it would be *very* helpful if NumPy explicitly had a table mapping each deprecation to what users are supposed to do to replace it.

Reply:
I remember that - I've got it on my list! 
Currently we have a [WIP list of what is being removed](https://github.com/numpy/numpy/issues/24306#issuecomment-1663564574) in details, and each relevant item from `remove` column will have a guideline (it will be placed in release notes) what should be used instead  
(by irrelevant item I mean e.g. some internal enums and vars that got exposed because of `from np.core import *` int top module)

Reply:
Ah, thanks for the clarification! Is it fair to say that `jnp.issubdtype` is a good replacement in most cases?

Reply:
Updated

Reply:
I think yes. Even simpler would be to recommend using a builtin `issubclass(arg1, arg2)` as that is the exact implementation of `issubsctype`, like: 
```
arr = np.array([1, 2, 3])
np.issubsctype(arr, np.integer) == issubclass(arr.dtype.type, np.integer)
```

Reply:
How is this different from `np.issubdtype(arr.dtype, np.integer)`?

Reply:
It ends up the same, also right recommendation.